### ISSUE TITLE
Unable to install >0.4.0 via npm

### ISSUE THREAD

**Issue by williamboman** [quick-lint/quick-lint-js#561]
Hello! I'm looking at adding quick-lint-js to https://github.com/williamboman/nvim-lsp-installer. I decided to use the npm distribution because it seemed to be the most consistent one across platforms. I'm however struggling to install any versions above `0.4.0`, where `npm install` yields the following error:

```
npm ERR! code 1
npm ERR! path /private/var/folders/ky/s6yyhm_d24d0jsrql4t8k4p40000gn/T/tmp.VpD5pcFbjt/node_modules/quick-lint-js
npm ERR! command failed
npm ERR! command sh -c node preinstall.js
npm ERR! node:internal/fs/utils:343
npm ERR!     throw err;
npm ERR!     ^
npm ERR!
npm ERR! Error: ENOENT: no such file or directory, copyfile '/private/var/folders/ky/s6yyhm_d24d0jsrql4t8k4p40000gn/T/tmp.VpD5pcFbjt/node_modules/quick-lint-js/darwin-arm64/bin/quick-lint-js' -> '/private/var/folders/ky/s6yyhm_d24d0jsrql4t8k4p40000gn/T/tmp.VpD5pcFbjt/node_modules/quick-lint-js/quick-lint-js.exe'
npm ERR!     at Object.copyFileSync (node:fs:2790:3)
npm ERR!     at Object.<anonymous> (/private/var/folders/ky/s6yyhm_d24d0jsrql4t8k4p40000gn/T/tmp.VpD5pcFbjt/node_modules/quick-lint-js/preinstall.js:23:4)
npm ERR!     at Module._compile (node:internal/modules/cjs/loader:1095:14)
npm ERR!     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1124:10)
npm ERR!     at Module.load (node:internal/modules/cjs/loader:975:32)
npm ERR!     at Function.Module._load (node:internal/modules/cjs/loader:816:12)
npm ERR!     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:79:12)
npm ERR!     at node:internal/main/run_main_module:17:47 {
npm ERR!   errno: -2,
npm ERR!   syscall: 'copyfile',
npm ERR!   code: 'ENOENT',
npm ERR!   path: '/private/var/folders/ky/s6yyhm_d24d0jsrql4t8k4p40000gn/T/tmp.VpD5pcFbjt/node_modules/quick-lint-js/darwin-arm64/bin/quick-lint-js',
npm ERR!   dest: '/private/var/folders/ky/s6yyhm_d24d0jsrql4t8k4p40000gn/T/tmp.VpD5pcFbjt/node_modules/quick-lint-js/quick-lint-js.exe'
npm ERR! }

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/williamboman/.npm/_logs/2021-12-06T15_35_35_663Z-debug.log
```

I'm running `Darwin Williams-MacBook-Air.local 21.2.0 Darwin Kernel Version 21.2.0: Wed Oct 20 00:23:21 PDT 2021; root:xnu-8019.60.40.0.1~26/RELEASE_ARM64_T8101 arm64 arm64 MacBookAir10,1 Darwin`

**Comment by strager** [comment]
What version of npm and Node.js do you have? The package successfully installs for me on a similar machine:
```
$ npm --version
7.3.0
$ node --version
v15.5.1
$ uname -a
Darwin strammer.lan 21.1.0 Darwin Kernel Version 21.1.0: Wed Oct 13 17:33:01 PDT 2021; root:xnu-8019.41.5~1/RELEASE_ARM64_T6000 arm64
$ sw_vers
ProductName:	macOS
ProductVersion:	12.0.1
BuildVersion:	21A559

$ sudo npm install --global --unsafe-perm quick-lint-js
Password:

added 1 package, and audited 2 packages in 668ms

1 package is looking for funding
  run `npm fund` for details

found 0 vulnerabilities
$ quick-lint-js --version
quick-lint-js version 0.7.0
```

**Comment by williamboman** [comment]
```
$ npm version
{
  npm: '8.0.0',
  node: '16.5.0',
  v8: '9.1.269.38-node.20',
  uv: '1.41.0',
  zlib: '1.2.11',
  brotli: '1.0.9',
  ares: '1.17.1',
  modules: '93',
  nghttp2: '1.42.0',
  napi: '8',
  llhttp: '6.0.2',
  openssl: '1.1.1k+quic',
  cldr: '39.0',
  icu: '69.1',
  tz: '2021a',
  unicode: '13.0',
  ngtcp2: '0.1.0-DEV',
  nghttp3: '0.1.0-DEV'
}
$ sw_vers
ProductName:    macOS
ProductVersion: 12.1
BuildVersion:   21C5021h
```

**Comment by strager** [comment]
I was able to reproduce the issue with Node.js version v16.13.0 and npm version 8.1.0.

**Comment by williamboman** [comment]
:+1: Btw for nvim-lsp-installer I went with the manual distribution instead, [like so](https://github.com/williamboman/nvim-lsp-installer/commit/1f6f518f752b9e71c5eaa0ed156bedea25f12a14). So this is not a blocker from my side anymore :)

**Comment by strager** [comment]
Thanks for reporting this issue! This problem is specific to macOS AArch64.

When I installed through Node.js version v15.5.1, quick-lint-js used the x86_64 executable. This is because my Node.js installation was x86_64 (running through Rosetta), not native AArch64.

In Node.js v16.13.0, `os.arch()` returns `"arm64"`. We use the result of `os.arch()` to pick a directory. But the directory `darwin-arm64` doesn't exist. `darwin-aarch64` *does* exist, but that's the wrong name.

I'll double-check older and newer versions of Node.js to make sure `os.arch()` consistently returns `"arm64"`. If it does, we just need to fix the name of the directory in the npm package.

**Comment by williamboman** [comment]
Yeah normalizing the arch is a tad bit annoying. Here's an [alias table](https://github.com/williamboman/nvim-lsp-installer/blob/main/lua/nvim-lsp-installer/platform.lua#L11-L19) I use. It's based on libuv, just like Node.js,

**Comment by strager** [comment]
Fixed in commit b054cfc4f6637c42db5a7f4eaa434f2161a3f7aa. I need to publish a new version to npm.

**Comment by strager** [comment]
Published [version 0.7.1 to npm](https://www.npmjs.com/package/quick-lint-js/v/0.7.1) with the fix.

Thanks again for reporting @williamboman!
